homeassistant/components/device_tracker/config_entry.py

```diff
@@ -164,6 +165,79 @@ def is_connected(self) -> bool:
         """Return true if the device is connected to the network."""
         raise NotImplementedError
 
+    @property
+    def unique_id(self) -> str | None:
+        """Return unique ID of the entity."""
+        return self.mac_address
+
+    @property
+    def device_info(self) -> DeviceInfo | None:
```

Comment:
Lets mark that final, so it would be catched by type checking

```suggestion
    @property
    @final
    def device_info(self) -> DeviceInfo | None:
```

Reply:
I had that at first. However Unifi also uses the ScannerEntity to integrate Unifi network devices, and for those it wants to include info.

Reply:
In the case of UniFi that wouldn't matter? As in, it could provide the devices in all other entities? The connection would be made in the same way as all other integrations at that point?

Reply:
Hmmm maybe you're right and it should still work but I'm too aggressive in cleaning up old devices. I only check if they have the config entry of the device tracker, but it could provide other entities. Let me check.

Reply:
Ok, doing it now once when setting up the first device tracker config entry